bot/cogs/trivia.py

```diff
@@ -0,0 +1,307 @@
+        elif category == 'geography' or category == 'geo':
+            category = 22
+        elif category == 'history':
+            category = 23
+        elif category == 'art':
+            category = 25
+        elif category == 'animals' or category == 'animal':
+            category = 27
+        elif category == 'comics' or category == 'comic':
+            category = 29
+        elif category == 'anime':
+            category = 31
+        elif category == 'cartoons' or category == 'cartoon':
+            category = 32
+
+        message = await ctx.send(embed=Embed(title='Loading.....', color=EMBED_COLOR))
+
+        while True:
+            t_data = await self.get_trivia(category)
+            embed = Embed(
+                title=t_data['question'],
+                description=f"1️⃣: {t_data['answers'][0]}\n\n"
+                            f"2️⃣: {t_data['answers'][1]}\n\n"
+                            f"3️⃣: {t_data['answers'][2]}\n\n"
+                            f"4️⃣: {t_data['answers'][3]}\n\n"
+                            f"You have 15 seconds or until {ctx.author.mention} answers!",
+                color=EMBED_COLOR
+            )
+            embed.set_author(name=f"{t_data['difficulty']} question for {t_data['points']} points")
+            embed.set_footer(text=t_data['category'])
+
+            await message.edit(
+                embed=embed
+            )
+            await message.add_reaction('1️⃣')
+            await message.add_reaction('2️⃣')
+            await message.add_reaction('3️⃣')
+            await message.add_reaction('4️⃣')
+
+            answers = {}
+
+            def r_check(reaction: Reaction, user: User) -> bool:
+                return reaction.message.id == message.id and not user.bot
+
+            while True:
+                try:
+                    reaction, user = await self.bot.wait_for('reaction_add', timeout=10.0, check=r_check)
+                except TimeoutError:
+                    break
+                if reaction.emoji == '1️⃣':
+                    answers[str(user.id)] = (1, user.name)
+                elif reaction.emoji == '2️⃣':
+                    answers[str(user.id)] = (2, user.name)
+                elif reaction.emoji == '3️⃣':
+                    answers[str(user.id)] = (3, user.name)
+                elif reaction.emoji == '4️⃣':
+                    answers[str(user.id)] = (4, user.name)
+
+                await reaction.remove(user)
+
+                if user.id == ctx.author.id:
+                    break
+
+            winners = []
+
+            for user in answers:
+                if answers[user][0] == t_data['correct']:
+                    if user in self.scores:
+                        self.scores[user][1] += t_data['points']
+                    else:
+                        self.scores[user] = [answers[user][1], t_data['points']]
+                    winners.append(answers[user][1])
+
+            with open("bot/scores.json", "w+") as file:
```

Comment:
this is a blocking file open/save that can happen quite often. I do not think it is worth changing for now, as this will be moved to the api. But something to be aware of. No fix needed for this at the moment.